src/Controller/Component/CaptchaComponent.php

```diff
@@ -252,7 +258,27 @@ private function __getSessionKey()
     {
         return $this->sessionKey;
     }
-    
+
+    function generate()
+    {
+        $old = $this->outputToBrowser;
+        $return = ['error' => true, 'data' => ''];
+        $this->outputToBrowser = false;
+        switch ($this->__getType()):
+            case 'image';
+                $return = ['error' => false, 'data' => $this->__imageCaptcha()];
```

Comment:
this could returns only data or 
other error could return here to handle in controller
so you can remove `$this->__setError(` from this component